### PR TITLE
Initialize telemetry client

### PR DESCRIPTION
**Pull request by MaggieKimani1**
Previously, the ` _telemetryClient` field in the `OpenApiService` and `ChangesService` classes weren't being initialized correctly.
To resolve this, I've created interfaces for both services and registered them in Startup as singletons.

### CODE REVIEW

File: ChangesService/Interfaces/IChangesService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+namespace ChangesService.Interfaces
+{
+    public interface IChangesService
```
Comment (baywet) [inline ChangesService/Interfaces/IChangesService.cs:7]: the interface should contain the methods prototypes so consumers can refer to the interface and don't need to know about the implementation/downcast. A PIE starts with abstraction :)

File: OpenAPIService/Interfaces/IOpenApiService.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+namespace OpenAPIService.Interfaces
+{
+    public interface IOpenApiService
```
Comment (baywet) [inline OpenAPIService/Interfaces/IOpenApiService.cs:7]: same as IChangeService

**Review by baywet — CHANGES_REQUESTED** [review]

File: ChangesService/Interfaces/IChangesService.cs (hunk 1 above)
Comment (nikithauc) [reply ChangesService/Interfaces/IChangesService.cs:7]: If `IChangeService` and `IOpenApiService` contain the same abstract methods then please maintain a single interface with  the prototype with the common method.

File: OpenAPIService/OpenApiService.cs [hunk 3]
```diff
@@ -203,15 +203,15 @@ public OpenApiDocument CreateFilteredDocument(OpenApiDocument source, string tit
                 }
                 else if (!_openApiRootNode.PathItems.ContainsKey(graphVersion))
                 {
-                    _openApiTraceProperties.Add(UtilityConstants.TelemetryPropertyKey_SanitizeIgnore, "GraphVersion");
+                    _openApiTraceProperties.Add(UtilityConstants.TelemetryPropertyKey_SanitizeIgnore, "OpenApiService");
```
Comment (baywet) [inline OpenAPIService/OpenApiService.cs:206]: leverage a constant instead? of nameof(OpenApiService)

**Review by baywet — CHANGES_REQUESTED** [review]

File: GraphExplorerPermissionsService/Services/PermissionsStore.cs [hunk 4]
```diff
@@ -86,9 +86,11 @@ private void SeedPermissionsTables()
 
             foreach (string permissionFilePath in _permissionsBlobNames)
             {
+                _permissionsTraceProperties.Add(UtilityConstants.TelemetryPropertyKey_SanitizeIgnore, "PermissionsStore");
```
Comment (baywet) [inline GraphExplorerPermissionsService/Services/PermissionsStore.cs:89]: constant or nameof

File: GraphWebApi/Controllers/GraphExplorerSamplesController.cs [hunk 5]
```diff
@@ -71,7 +71,7 @@ public async Task<IActionResult> GetSampleQueriesListAsync(string search, string
                     return NotFound();
                 }
 
-                _samplesTraceProperties.Add(UtilityConstants.TelemetryPropertyKey_Count, "SamplesCount");
+                _samplesTraceProperties.Add(UtilityConstants.TelemetryPropertyKey_SanitizeIgnore, "SamplesCount");
```
Comment (baywet) [inline GraphWebApi/Controllers/GraphExplorerSamplesController.cs:74]: constant or nameof

File: OpenAPIService.Test/OpenAPIDocumentCreatorMock.cs [hunk 6]
```diff
@@ -13,9 +14,15 @@ namespace OpenAPIService.Test
...
     /// </summary>
-    public static class OpenAPIDocumentCreatorMock
+    public class OpenAPIDocumentCreatorMock
     {
         private static readonly ConcurrentDictionary<string, OpenApiDocument> _OpenApiDocuments = new ConcurrentDictionary<string, OpenApiDocument>();
+        private readonly IOpenApiService _openApiService;
+
+        public OpenAPIDocumentCreatorMock(IOpenApiService openApiService)
+        {
+            _openApiService = openApiService;
```
Comment (baywet) [inline OpenAPIService.Test/OpenAPIDocumentCreatorMock.cs:24]: defensive programing otherwise the CreateOpenApiDocument method might fail

Comment (MaggieKimani1) [reply OpenAPIService.Test/OpenAPIDocumentCreatorMock.cs:24]: Let me add that..

**Review by baywet — APPROVED** [review]

**Pull request merged**